### PR TITLE
Fix healing bucket properly

### LINKED ISSUE #6710: Partially existing buckets, listed, but not accessible or removable, #6703 related...
For some reason that i dont know exactly, a single bucket im my cluster have not been deleted completely, after a delete operation he still remained on 33% of the disks, maybe due to network instability...

Whe i try to re-delete it with:

mc rm server1/bucketerror --force
mc: <ERROR> Failed to remove `srvminio1/bucketerror`. Bucket `bucketerror` does not exist.

But when a list it with:
mc ls server1/
[2018-10-26 11:30:56 UTC]     0B bucketerror/

The buckt is in the list...

## Expected Behavior
I dont know what is the right behavior in this case, since the bucket was removed from API, i think the right action in this case was to:
First: not list the bucket anymore, even if he still exists in some disk/server
Second: maybe auto remove this bucket when i run a heal command

## Current Behavior
Bucket appear in list command but i am not able to remove it or do anything with it

## Possible Solution
Maybe the quorum checking on DELETE Bucket operations can be relaxed...

This error is related to #6703

## Steps to Reproduce (for bugs)
Create a bucket , remove it manally in the disks to less than quorun...

## Your Environment
* Version used (`minio version`): Lastest

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Bucket should be healed properly if it partially exists
on only one set, since bucket is common for all sets.

## Motivation and Context
Fixes #6710

## Regression
No

## How Has This Been Tested?
Using `mc admin heal -r --json`, you need to create this issue manually using following script

```
#!/bin/bash

export MINIO_ERASURE_SET_DRIVE_COUNT=6
export MINIO_ACCESS_KEY="minio"
export MINIO_SECRET_KEY="minio123"

for i in {01..24}; do
    minio server --address ":90${i}" https://127.0.0.1:9001/tmp/export01 https://127.0.0.1:9002/tmp/export02 https://127.0.0.1:9003/tmp/export03 https://127.0.0.1:9004/tmp/export04 https://127.0.0.1:9005/tmp/export05 https://127.0.0.1:9006/tmp/export06 https://127.0.0.1:9007/tmp/export07 https://127.0.0.1:9008/tmp/export08 https://127.0.0.1:9009/tmp/export09 https://127.0.0.1:9010/tmp/export10 https://127.0.0.1:9011/tmp/export11 https://127.0.0.1:9012/tmp/export12 https://127.0.0.1:9013/tmp/export13 https://127.0.0.1:9014/tmp/export14 https://127.0.0.1:9015/tmp/export15 https://127.0.0.1:9016/tmp/export16 https://127.0.0.1:9017/tmp/export17 https://127.0.0.1:9018/tmp/export18 https://127.0.0.1:9019/tmp/export19 https://127.0.0.1:9020/tmp/export20 https://127.0.0.1:9021/tmp/export21 https://127.0.0.1:9022/tmp/export22 https://127.0.0.1:9023/tmp/export23 https://127.0.0.1:9024/tmp/export24 &
done
```

```
mc mb myminio/testbucket
```

Now take down all servers and delete bucket from only 2 sets `rmdir /tmp/export{01..16}` , restart all the servers again. 

Now try to do `mc admin heal -r --json` - server would return a wrong error with quorum missing, we don't need to wait for this instead we should heal properly which is what this PR does.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

File: cmd/xl-v1-healing.go [hunk 1]
```diff
@@ -67,13 +67,6 @@ func (xl xlObjects) HealBucket(ctx context.Context, bucket string, dryRun bool)
 
 	storageDisks := xl.getDisks()
 
-	// Check if bucket doesn't exist in writeQuorum number of disks, if quorum
-	// number of disks returned that bucket does not exist we quickly return
-	// and do not proceed to heal.
-	if err = checkBucketExistsInQuorum(ctx, storageDisks, bucket); err != nil {
```
Comment (kannappanr) [inline cmd/xl-v1-healing.go:73]: `checkBucketExistsInQuorum` function can also be removed.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Review by kannappanr — APPROVED** [review]
LGTM

**Review by poornas — APPROVED** [review]
LGTM

**Comment by tnatanael** [comment]
It works thank's @harshavardhana , i found that if i create folders manually on the exports/vols, since there is no metadata, the heal crashes again with insuficient write quorum, maybe this is the right action, but not the correct error message, what you think?

**Pull request merged**